atef/widgets/config/run_base.py

```diff
@@ -0,0 +1,416 @@
+        # overall result label
+        if not self.data:
+            logger.warning('No config associated with this step')
+            return
+        self.update_icon(self.result_label, self.results)
+        self.update_label_tooltip(self.result_label, self.results)
+
+        # step result
+        if self.step_results:
+            self.update_icon(self.run_success_label, self.step_results)
+            self.update_label_tooltip(self.run_success_label, self.step_results)
+
+            self.update_icon(self.verify_label, self.verify_results)
+            self.update_label_tooltip(self.verify_label, self.verify_results)
+
+    def update_icon(self, label: QLabel, results: List[Result]) -> None:
+        """ Helper method to update icon on ``label`` based on ``results`` """
+        combined_step_result = combine_results(results)
+
+        chosen_icon = self.style_icons[combined_step_result.severity]
+        icon = self.style().standardIcon(chosen_icon)
+
+        label.setPixmap(icon.pixmap(25, 25))
+
+    def update_label_tooltip(self, label: QLabel, results: List[Result]) -> None:
+        """ Helper method to update tooltip for ``label`` based on ``results`` """
+        tt = ''
+        for r in results:
+            uni_icon = self.unicode_icons[r.severity]
+            tt += f'{uni_icon}: {r.reason or "-"}<br>'
+
+        label.setToolTip('<p>' + tt.rstrip('<br>') + '</p>')
+
+    def update_all_icons_tooltips(self) -> None:
+        """ Convenience method for updating all the icons and tooltips """
+        self.update_icon(self.result_label, self.results)
+        self.update_label_tooltip(self.result_label, self.results)
+
+        # Extras for active checkouts
+        if self.step_results:
+            self.update_icon(self.run_success_label, self.step_results)
+            self.update_label_tooltip(self.run_success_label, self.step_results)
+
+        if self.verify_results:
+            self.update_icon(self.verify_label, self.verify_results)
+            self.update_label_tooltip(self.verify_label, self.verify_results)
+
+    def event(self, event: QtCore.QEvent) -> bool:
+        """ Overload event method to update tooltips on tooltip-request """
+        # Catch tooltip events to update status tooltip
+        if event.type() == QtCore.QEvent.ToolTip:
+            self.update_all_icons_tooltips()
+        return super().event(event)
+
+    @property
+    def results(self) -> List[Result]:
+        return [c.result for c in self.data]
+
+    @property
+    def step_results(self) -> List[Result]:
+        try:
+            return [c.step_result for c in self.data]
+        except AttributeError:
+            return None
+
+    @property
+    def verify_results(self) -> List[Result]:
+        try:
+            return [c.verify_result for c in self.data]
+        except AttributeError:
+            return None
+
+    def setup_next_button(self, next_item=None) -> None:
```

Comment:
I think something is incomplete or buggy in the next button

![image](https://user-images.githubusercontent.com/10647860/224184724-1874c817-bb10-456d-847f-2ad95c245ea0.png)

Reply:
I figured out why this happened about an hour ago and only just managed to fix it.  Qt being rascally as usual